jobs/pkg/kube/secret.go

```diff
@@ -60,9 +63,17 @@ func GetExistingValidTlsSecret(ctx context.Context, kube kubernetes.Interface, s
 		if now.Before(cert.NotBefore) || now.After(cert.NotAfter) {
 			return nil, nil
 		}
+
+		// check if the cert is valid for this service
+		if certgen.ValidForService(cert.DNSNames, svcName, svcNamespace) {
```

Comment:
So as long as any cert is valid for our service then we are happy? 
If thats true then can we short circuit the dns checks for certs after? 

```suggestion
		if !matchesSVC && certgen.ValidForService(cert.DNSNames, svcName, svcNamespace) {
```

Reply:
ah yeah good point, fixed in 44783a4